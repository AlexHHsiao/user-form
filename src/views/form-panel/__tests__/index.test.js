import renderer from 'react-test-renderer';
import FormPanel from '../index';

describe("FormPanel Tests", () => {
    it('Matches DOM Snapshot', () => {
        const tree = renderer
            .create(<FormPanel setUpdateData={jest.fn} onSubmit={jest.fn}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

