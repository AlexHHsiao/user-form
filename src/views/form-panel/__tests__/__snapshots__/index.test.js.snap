// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPanel Tests Matches DOM Snapshot 1`] = `
<div
  className="css-1wfgkgb"
>
  <div
    className="css-qzib2e"
  >
    Add User
  </div>
  <form
    className="css-18hfe0s"
    onChange={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="css-1hrz0hu"
    >
      <div
        className="css-1l7cisx"
      >
        <label
          className="css-1gk38rd"
        >
          First Name
        </label>
      </div>
      <div
        className="css-1l7cisx"
      >
        <input
          className="css-vx6d1k"
          name="firstName"
          pattern="[A-Za-z0-9]{1,20}"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="css-1hrz0hu"
    >
      <div
        className="css-1l7cisx"
      >
        <label
          className="css-1gk38rd"
        >
          Last Name
        </label>
      </div>
      <div
        className="css-1l7cisx"
      >
        <input
          className="css-vx6d1k"
          name="lastName"
          pattern="[A-Za-z0-9]{1,20}"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="css-1hrz0hu"
    >
      <div
        className="css-1l7cisx"
      >
        <label
          className="css-1gk38rd"
        >
          Email
        </label>
      </div>
      <div
        className="css-1l7cisx"
      >
        <input
          className="css-vx6d1k"
          name="email"
          pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="css-1hrz0hu"
    >
      <div
        className="css-1l7cisx"
      >
        <label
          className="css-1gk38rd"
        >
          Status
        </label>
      </div>
      <div
        className="css-1l7cisx"
      >
        <select
          className="css-vx6d1k"
          name="status"
          value="active"
        >
          <option
            value="active"
          >
            Active
          </option>
          <option
            value="inactive"
          >
            Inactive
          </option>
          <option
            value="disabled"
          >
            Disabled
          </option>
        </select>
      </div>
    </div>
    <div
      className="css-1t4ef8e"
    >
      <button
        className="css-e4ymff"
        type="submit"
      >
        Add
      </button>
    </div>
    <div
      className="css-1t4ef8e"
    >
      <button
        className="css-e4ymff"
        type="reset"
      >
        Reset
      </button>
    </div>
  </form>
</div>
`;

exports[`LandingPage Tests Matches DOM Snapshot 1`] = `
Array [
  <div
    className="css-q2igk3"
  >
    <span
      aria-label="check-circle"
      className="anticon anticon-check-circle"
      role="img"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    >
      <svg
        aria-hidden="true"
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
        />
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
      </svg>
    </span>
  </div>,
  <div
    className="css-1hrz0hu"
  >
    <div
      className="css-3ocuqv"
    >
      <div
        className="css-1wfgkgb"
      >
        <div
          className="css-qzib2e"
        >
          Add User
        </div>
        <form
          className="css-18hfe0s"
          onChange={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="css-1hrz0hu"
          >
            <div
              className="css-1l7cisx"
            >
              <label
                className="css-1gk38rd"
              >
                First Name
              </label>
            </div>
            <div
              className="css-1l7cisx"
            >
              <input
                className="css-vx6d1k"
                name="firstName"
                pattern="[A-Za-z0-9]{1,20}"
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-1hrz0hu"
          >
            <div
              className="css-1l7cisx"
            >
              <label
                className="css-1gk38rd"
              >
                Last Name
              </label>
            </div>
            <div
              className="css-1l7cisx"
            >
              <input
                className="css-vx6d1k"
                name="lastName"
                pattern="[A-Za-z0-9]{1,20}"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-1hrz0hu"
          >
            <div
              className="css-1l7cisx"
            >
              <label
                className="css-1gk38rd"
              >
                Email
              </label>
            </div>
            <div
              className="css-1l7cisx"
            >
              <input
                className="css-vx6d1k"
                name="email"
                pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
                required={true}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="css-1hrz0hu"
          >
            <div
              className="css-1l7cisx"
            >
              <label
                className="css-1gk38rd"
              >
                Status
              </label>
            </div>
            <div
              className="css-1l7cisx"
            >
              <select
                className="css-vx6d1k"
                name="status"
                value="active"
              >
                <option
                  value="active"
                >
                  Active
                </option>
                <option
                  value="inactive"
                >
                  Inactive
                </option>
                <option
                  value="disabled"
                >
                  Disabled
                </option>
              </select>
            </div>
          </div>
          <div
            className="css-1t4ef8e"
          >
            <button
              className="css-e4ymff"
              type="submit"
            >
              Add
            </button>
          </div>
          <div
            className="css-1t4ef8e"
          >
            <button
              className="css-e4ymff"
              type="reset"
            >
              Reset
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="css-s5b1p8"
    >
      <div
        className="css-14z6jlm"
      >
        <div
          className="css-qzib2e"
        >
          User List
        </div>
        <div
          className="css-16w8qzd"
        >
          <div
            className="css-9fupyx"
          >
            <div
              className="css-1715rqs"
            >
              <div
                className="css-1sceoeu"
              >
                <div
                  className="css-1r1cjz"
                >
                  John
                   
                  Schmitt
                </div>
                <div
                  className="css-1fxqjq"
                >
                  <a
                    href="mailto:johnschmitt@example.com"
                  >
                    johnschmitt@example.com
                  </a>
                </div>
                <div
                  className="css-19ay2nh"
                >
                  <div
                    className="css-1f47qz5"
                    type="success"
                  >
                    <span
                      aria-label="check-circle"
                      className="anticon anticon-check-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="check-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                      </svg>
                    </span>
                     
                    active
                  </div>
                </div>
                <div
                  className="css-1r1cjz"
                >
                  Created: 
                  01/05/2020
                </div>
                <div
                  className="css-1r1cjz"
                >
                  Updated: 
                  01/05/2020
                </div>
                <div
                  className="css-1r1cjz"
                >
                  <div
                    className="css-1wbxo26"
                  >
                    <button
                      className="css-s4gat3"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="edit"
                        className="anticon anticon-edit"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="edit"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      className="css-s4gat3"
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="delete"
                        className="anticon anticon-delete"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="delete"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="css-1scet2y"
            >
              <div
                className="css-1sceoeu"
              >
                <div
                  className="css-1r1cjz"
                >
                  Beth
                   
                  Forest
                </div>
                <div
                  className="css-1fxqjq"
                >
                  <a
                    href="mailto:bethforest@example.com"
                  >
                    bethforest@example.com
                  </a>
                </div>
                <div
                  className="css-19ay2nh"
                >
                  <div
                    className="css-1nwyet0"
                    type="error"
                  >
                    <span
                      aria-label="close-circle"
                      className="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                        />
                        <path
                          d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                      </svg>
                    </span>
                     
                    inactive
                  </div>
                </div>
                <div
                  className="css-1r1cjz"
                >
                  Created: 
                  06/06/2021
                </div>
                <div
                  className="css-1r1cjz"
                >
                  Updated: 
                  06/06/2022
                </div>
                <div
                  className="css-1r1cjz"
                >
                  <div
                    className="css-1wbxo26"
                  >
                    <button
                      className="css-s4gat3"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="edit"
                        className="anticon anticon-edit"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="edit"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      className="css-s4gat3"
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="delete"
                        className="anticon anticon-delete"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="delete"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="css-1scet2y"
            >
              <div
                className="css-1sceoeu"
              >
                <div
                  className="css-1r1cjz"
                >
                  Edward
                   
                  Jones
                </div>
                <div
                  className="css-1fxqjq"
                >
                  <a
                    href="mailto:edwardjones@example.com"
                  >
                    edwardjones@example.com
                  </a>
                </div>
                <div
                  className="css-19ay2nh"
                >
                  <div
                    className="css-1e5udie"
                    type="default"
                  >
                    <span
                      aria-label="stop"
                      className="anticon anticon-stop"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="stop"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372 0-89 31.3-170.8 83.5-234.8l523.3 523.3C682.8 852.7 601 884 512 884zm288.5-137.2L277.2 223.5C341.2 171.3 423 140 512 140c205.4 0 372 166.6 372 372 0 89-31.3 170.8-83.5 234.8z"
                        />
                      </svg>
                    </span>
                     
                    disabled
                  </div>
                </div>
                <div
                  className="css-1r1cjz"
                >
                  Created: 
                  09/24/2022
                </div>
                <div
                  className="css-1r1cjz"
                >
                  Updated: 
                  11/03/2022
                </div>
                <div
                  className="css-1r1cjz"
                >
                  <div
                    className="css-1wbxo26"
                  >
                    <button
                      className="css-vckfoy"
                      disabled={true}
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="edit"
                        className="anticon anticon-edit"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="edit"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      className="css-s4gat3"
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        aria-label="delete"
                        className="anticon anticon-delete"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="delete"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
